### ISSUE TITLE
Toggle component no longer toggles after the first toggle

### ISSUE THREAD

**Issue by petevk** [smooth-code/smooth-ui#138]
## 🐛 Bug Report

The `Toggler` component doesn't seem to allow for toggling after it's been clicked once. I was using smooth-ui 7.x and it would allow the `onToggle` button to be clicked more than once, but after upgrading to 10.x it hasn't worked anymore.

This bug is also present in the living docs [here](https://smooth-ui.smooth-code.com/docs-utilities-toggler).

I know the render prop signature was changed in 10.x to return an array of `[toggled, onToggle]` instead of the object syntax, and I've updated the example to use that instead.

## To Reproduce

Click the toggle button a bunch of times. The first time it's clicked, it's toggled, but additional clicks does not toggle.

## Link to repl or repo (highly encouraged)

This issue is present in the docs (https://smooth-ui.smooth-code.com/docs-utilities-toggler), but I've also created a minimal code sandbox:
https://codesandbox.io/s/fork-7lynh

Thanks! Let me know if you need anymore information from me.

**Comment by petevk** [comment]
Alright I think I see what happened.

In my codesandbox and the docs, the toggler component is created like this:

```js
<Toggler>
  {([toggled, onToggle]) => (
    <Button onClick={onToggle}>{toggled ? "ON" : "OFF"}</Button>
  )}
</Toggler>
```

This doesn't work like it did in 9.x and below. However, this **does** work:


```js
<Toggler>
  {([toggled, onToggle]) => (
    <Button onClick={() => onToggle()}>{toggled ? "ON" : "OFF"}</Button>
  )}
</Toggler>
```
(notice the difference in the `onClick` function)

This changed because of [this line](https://github.com/smooth-code/smooth-ui/blob/v9.1.0/packages/shared/core/Toggler.js#L21) in 9.x, which checked to see if the type of value is a boolean before toggling. In 10.x, [this line](https://github.com/smooth-code/smooth-ui/blob/v10.1.0/packages/shared/core/hooks/useToggle.js#L11) checks to see if it's undefined and doesn't check the type. Since `onClick`s are called with their event as the argument, `toggled` is set to an event.

I'd argue that this is a bug, since I think the Toggler component really only makes sense for booleans, right?

**Comment by neoziro** [comment]
Since event is considered as `true`,  the actual behaviour looks correct to me. But maybe we should test if it is a boolean to avoid this kind of errors. So can you please add a check on boolean in a PR?

**Comment by petevk** [comment]
> Since event is considered as true, the actual behaviour looks correct to me. But maybe we should test if it is a boolean to avoid this kind of errors. So can you please add a check on boolean in a PR?

The event is considered `true`, but since it's being set to true on every onclick it means it never toggles.

I'll see if I can get something up later today. Thanks!

**Comment by neoziro** [comment]
Closed by #139
